shared/logger.go

```diff
@@ -143,14 +135,31 @@ func newAppEngineFlexContext(r *http.Request, project string, commonResource *mr
 // commonResource is an optional override to the monitored resource details appended to each log.
 // e.g. in the Flex environment, it pays to override this value to type gae_app, to ensure finding
 // logs is consistent between services.
-func HandleWithGoogleCloudLogging(h http.HandlerFunc, project string, commonResource *mrpb.MonitoredResource) http.HandlerFunc {
+func HandleWithGoogleCloudLogging(h http.HandlerFunc, gcClient *gclog.Client, project string, commonResource *mrpb.MonitoredResource) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		ctx, err := newAppEngineFlexContext(r, project, commonResource)
+		// See https://cloud.google.com/appengine/docs/flexible/go/writing-application-logs
+		traceID := strings.Split(r.Header.Get("X-Cloud-Trace-Context"), "/")[0]
+		if traceID != "" {
+			traceID = fmt.Sprintf("projects/%s/traces/%s", project, traceID)
+		}
+		ctx, err := newAppEngineFlexContext(r, gcClient, traceID, commonResource)
 		if err != nil {
 			h(w, r)
 			return
 		}
 		h(w, r.WithContext(ctx))
+		parentLogger := gcClient.Logger("request", gclog.CommonResource(commonResource))
+		e := gclog.Entry{
+			Trace:    traceID,
+			Severity: gclog.Info,
+			HTTPRequest: &gclog.HTTPRequest{
+				Request: r,
```

Comment:
Let's try setting the latency here (by getting a timestamp at the beginning of the function and subtracting it from the current timestamp). Looks like this might be the missing piece to group logs correctly.

Reply:
Right, it is mentioned in [the issue you found](https://github.com/googleapis/google-cloud-go/issues/720#issuecomment-320780311)